src/lib/operators/partitioning.cpp

```diff
@@ -0,0 +1,301 @@
+                          casted_target->null_values().begin() + target_index);
+              target_index++;
+            }
+          }
+        } else {
+          for (const auto null_value : casted_source->null_values()) {
+            Assert(!null_value, "Trying to insert NULL into non-NULL column");
+          }
+        }
+      }
+    } else if (auto casted_dummy_source = std::dynamic_pointer_cast<const ValueColumn<int32_t>>(source)) {
+      // We use the column type of the Dummy table used to insert a single null value.
+      // A few asserts are needed to guarantee correct behaviour.
+      Assert(rows_to_copy.size() == 1, "Cannot insert multiple unknown null values at once.");
+      Assert(casted_dummy_source->size() == 1, "Source column is of wrong type.");
+      Assert(casted_dummy_source->null_values().front() == true, "Only value in dummy table must be NULL!");
+      Assert(target_is_nullable, "Cannot insert NULL into NOT NULL target.");
+
+      // Ignore source value and only set null to true
+      casted_target->null_values()[target_start_index] = true;
+    } else {
+      // } else if(auto casted_source = std::dynamic_pointer_cast<ReferenceColumn>(source)){
+      // since we have no guarantee that a referenceColumn references only a single other column,
+      // this would require us to find out the referenced column's type for each single row.
+      // instead, we just use the slow path below.
+
+      size_t target_index = target_start_index;
+      for (size_t row_to_copy : rows_to_copy) {
+        if (row_to_copy >= source_start_index) {
+          auto ref_value = (*source)[row_to_copy];
+          if (variant_is_null(ref_value)) {
+            Assert(target_is_nullable, "Cannot insert NULL into NOT NULL target");
+            values[target_index] = T{};
+            casted_target->null_values()[target_index] = true;
+          } else {
+            values[target_index] = type_cast<T>(ref_value);
+          }
+          target_index++;
+        }
+      }
+    }
+  }
+};
+
+Partitioning::Partitioning(const std::string& table_to_partition_name,
+                           std::shared_ptr<AbstractPartitionSchema> target_partition_schema)
+    : AbstractReadWriteOperator(),
+      _table_to_partition_name{table_to_partition_name},
+      _target_partition_schema{target_partition_schema} {}
+
+const std::string Partitioning::name() const { return "Partitioning"; }
+
+std::shared_ptr<const Table> Partitioning::_on_execute(std::shared_ptr<TransactionContext> context) {
+  const auto table_to_partition = _get_table_to_be_partitioned();
+  _lock_table(table_to_partition);
+  auto partitioned_table = _create_partitioned_table_copy(table_to_partition);
+  _copy_table_content(table_to_partition, partitioned_table);
+  _replace_table(partitioned_table);
+
+  return nullptr;
+}
+
+std::shared_ptr<Table> Partitioning::_get_table_to_be_partitioned() {
+  return StorageManager::get().get_table(_table_to_partition_name);
```

Comment:
No need to add a method for a one-liner that is only used once.

Reply:
The method name is a description of what happens here on a higher level of abstraction. The compiler will inline this anyway.